compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal

```diff
@@ -1069,6 +1088,14 @@ function generateDefaultConstructor(jvm:ClassWriter cw, string ownerClass) {
     mv.visitEnd();
 }
 
+function generateDiagnosticPos(bir:DiagnosticPos pos, jvm:MethodVisitor mv) {
+    if (pos.sLine != -1) {
```

Comment:
Why this check is needed? When can it be -1?

Reply:
We have Goto  terminators without position hence this will have default values -1. These terminators will be skipped during code gen.